crates/settlement-clients/ethereum/src/tests/mod.rs

```diff
@@ -0,0 +1,341 @@
+use std::path::PathBuf;
+use std::time::Duration;
+use std::{
+    fs::{self, File},
+    io::BufReader,
+    str::FromStr,
+};
+use tokio::time::sleep;
+use utils::env_utils::get_env_var_or_panic;
+
+use settlement_client_interface::SettlementClient;
+
+use crate::conversion::to_padded_hex;
+use crate::EthereumSettlementClient;
+use alloy::providers::Provider;
+use alloy_primitives::FixedBytes;
+
+// Using the Pipe trait to write chained operations easier
+trait Pipe: Sized {
+    fn pipe<T, F: FnOnce(Self) -> T>(self, f: F) -> T {
+        f(self)
+    }
+}
+
+// Implement Pipe for all types
+impl<S> Pipe for S {}
+
+// TODO: betterment of file routes
+
+use lazy_static::lazy_static;
+use url::Url;
+
+lazy_static! {
+    static ref ENV_FILE_PATH: PathBuf = PathBuf::from(".env.test");
+    static ref CURRENT_PATH: String = env::current_dir()
+        .expect("Failed to get current directory")
+        .to_str()
+        .expect("Path contains invalid Unicode")
+        .to_string();
+    static ref ETH_RPC: String = get_env_var_or_panic("ETHEREUM_BLAST_RPC_URL");
+    pub static ref STARKNET_OPERATOR_ADDRESS: Address =
+        Address::from_str("0x2C169DFe5fBbA12957Bdd0Ba47d9CEDbFE260CA7").expect("Unable to parse address");
+    static ref STARKNET_CORE_CONTRACT_ADDRESS: Address =
+        Address::from_str("0xc662c410c0ecf747543f5ba90660f6abebd9c8c4").expect("Could not impersonate account.");
+    pub static ref TEST_NONCE: u64 = 666068;
+}
+
+// SOLIDITY FUNCTIONS NEEDED
+sol!(
+    #[allow(missing_docs)]
+    #[sol(rpc)]
+    STARKNET_CORE_CONTRACT,
+    "src/test_data/contract_abi/starknet_core_contract.json"
+);
+
+sol! {
+    #[allow(missing_docs)]
+    #[sol(rpc, bytecode="6080604052348015600e575f80fd5b506101c18061001c5f395ff3fe608060405234801561000f575f80fd5b5060043610610029575f3560e01c8063b72d42a11461002d575b5f80fd5b6100476004803603810190610042919061010d565b610049565b005b50505050565b5f80fd5b5f80fd5b5f80fd5b5f80fd5b5f80fd5b5f8083601f84011261007857610077610057565b5b8235905067ffffffffffffffff8111156100955761009461005b565b5b6020830191508360208202830111156100b1576100b061005f565b5b9250929050565b5f8083601f8401126100cd576100cc610057565b5b8235905067ffffffffffffffff8111156100ea576100e961005b565b5b6020830191508360018202830111156101065761010561005f565b5b9250929050565b5f805f80604085870312156101255761012461004f565b5b5f85013567ffffffffffffffff81111561014257610141610053565b5b61014e87828801610063565b9450945050602085013567ffffffffffffffff81111561017157610170610053565b5b61017d878288016100b8565b92509250509295919450925056fea2646970667358221220fa7488d5a2a9e6c21e6f46145a831b0f04fdebab83868dc2b996c17f8cba4d8064736f6c634300081a0033")]
+    contract DummyCoreContract {
+        function updateStateKzgDA(uint256[] calldata programOutput, bytes calldata kzgProof)  external {
+        }
+    }
+}
+
+struct EthereumTestBuilder {
+    fork_block: Option<u64>,
+    impersonator: Option<Address>,
+}
+
+struct EthereumTest {
+    _anvil: AnvilInstance,
```

Comment:
Why are you prefixing this with an underscore?

Reply:
So, 
We notices the an `anvil` server is needed to be in scope in any function that is using RPC calls to `anvil`.

We create `anvil` server in `EthereumTestBuilder` if we don't pass it to `update_state_blob_with_dummy_contract_works` we get `ConnectionRefused` Error suggesting that the server that was started in `EthereumTestBuilder` has now been closed.

I believe this has to do with how `rust` handles `variable scoping`.

So, `_anvil` as a variable is never used, it's only there to be in the scope, hence the `_`.

This is my understanding, please feel free to correct it / broaden the perspective / suggest an alternative & optimised way to achieve the same.